integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -264,8 +305,116 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
           s"Carbon Implicit column ${col.column} is not allowed in" +
           s" column name while creating table")
       }
+    }
+  }
 
+  /**
+   * The method parses, validates and processes the index_handler property.
+   * @param tableProperties Table properties
+   * @param tableFields  Sequence of table fields
+   * @return <Seq[Field]> Sequence of index fields to add to table fields
+   */
+  private def processIndexProperty(tableProperties: mutable.Map[String, String],
```

Comment:
why cannot we  move these validations to respective handlers?

Reply:
Moved to CarbonParserUtil similar to other parse util methods.